tool/tctl/common/plugins_command.go

```diff
@@ -96,11 +189,106 @@ func (p *PluginsCommand) Cleanup(ctx context.Context, clusterAPI *auth.Client) e
 	return nil
 }
 
+// InstallSCIM implements `tctl plugins install scim`, installing a SCIM integration
+// plugin into the teleport cluster
+func (p *PluginsCommand) InstallSCIM(ctx context.Context, client *auth.Client) error {
+	log := p.config.Logger.With(logFieldPlugin, p.install.name)
+
+	log.DebugContext(ctx, "Fetching cluster info...")
+	info, err := client.Ping(ctx)
+	if err != nil {
+		log.ErrorContext(ctx, "Failed fetching cluster info", logErrorMessage(err))
+		return trace.Wrap(err)
+	}
+
+	scimTokenHash, err := bcrypt.GenerateFromPassword([]byte(p.install.scim.token), bcrypt.DefaultCost)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	connectorID := p.install.scim.samlConnector
+	log.DebugContext(ctx, "Validating SAML Connector...", logFieldSAMLConnector, connectorID)
+	connector, err := client.GetSAMLConnector(ctx, p.install.scim.samlConnector, false)
+	if err != nil {
+		log.ErrorContext(ctx, "Failed validating SAML connector",
+			slog.String(logFieldSAMLConnector, connectorID),
+			logErrorMessage(err))
+
+		if !p.install.scim.force {
+			return trace.Wrap(err)
+		}
+	}
+
+	role := p.install.scim.role
+	log.DebugContext(ctx, "Validating Default Role...", logFieldRole, role)
+	if _, err := client.GetRole(ctx, role); err != nil {
+		log.ErrorContext(ctx, "Failed validating role", slog.String(logFieldRole, role), logErrorMessage(err))
+		if !p.install.scim.force {
+			return trace.Wrap(err)
+		}
+	}
+
+	request := &pluginsv1.CreatePluginRequest{
+		Plugin: &types.PluginV1{
+			SubKind: types.PluginSubkindAccess,
+			Metadata: types.Metadata{
+				Labels: map[string]string{
+					types.HostedPluginLabel: "true",
+				},
+				Name: p.install.name,
+			},
+			Spec: types.PluginSpecV1{
+				Settings: &types.PluginSpecV1_Scim{
+					Scim: &types.PluginSCIMSettings{
+						SamlConnectorName: p.install.scim.samlConnector,
+						DefaultRole:       p.install.scim.role,
+					},
+				},
+			},
+		},
+		StaticCredentials: &types.PluginStaticCredentialsV1{
+			ResourceHeader: types.ResourceHeader{
+				Metadata: types.Metadata{
+					Name: p.install.name,
```

Comment:
just double checking that this is supposed to be the same name as the `plugin.metadata.name`?